logic.js

```diff
@@ -0,0 +1,176 @@
+
+// Outer functions (to be reused as if they were inheretend from a parent class)
+// remove element from the array
+const removeElementById = function(array, id) {
+    console.log('Before iterating over the array:')
+            for (let m=0; m<array.length; m++){
+                console.log(array[m].getId())
+            }
+    console.log('While iterating over the array:')
+    for(let i=0; i<array.length; i++){
+        if(array[i].getId() == id){
+            // splice function does not remove the last item in the mutated array, that's why we have to reassign an empty array
+            if(array.length == 1){
+                array = []
+                break
+            }
+            console.log('the element that should be removed is of id: '+array[i].getId())
+            console.log('i: '+i)
+            let deleted = array.splice(i, 1)
+            
```

Comment:
remove all the "development" logic (aka all the logging) and make the code nice and short, just for the purpose of deleting an item from the array